src/plugin/parsers/light_parser.cpp

```diff
@@ -0,0 +1,260 @@
+#include <maya/MFnSpotLight.h>
+#include <maya/MFnTransform.h>
+#include <maya/MGlobal.h>
+#include <maya/MQuaternion.h>
+
+#include <DirectXMath.h>
+
+using namespace DirectX;
+
+namespace wr
+{
+	//template std::shared_ptr<LightNode> SceneGraph::CreateChild( std::shared_ptr<Node> const & parent, LightType tid, DirectX::XMVECTOR col );
+	
+}
+// region for internally used functions, these functions cannot be use outside this cpp file
+#pragma region INTERNAL_FUNCTIONS
+
+static void updateTransform( MFnTransform& transform, std::shared_ptr<wr::LightNode> mesh_node )
+{
+	MStatus status = MS::kSuccess;
+
+	MVector pos = transform.getTranslation( MSpace::kTransform, &status );
+
+	MQuaternion qrot;
+	status = transform.getRotation( qrot, MSpace::kTransform );
+	qrot.normalizeIt();
+	MEulerRotation rot = qrot.asEulerRotation();
+	rot.reorderIt( MEulerRotation::kZXY );
+
+	double3 scale;
+	status = transform.getScale( scale );
+
+	assert( status == MS::kSuccess );
+
+	mesh_node->SetPosition( { static_cast< float >( pos.x ), static_cast< float >( pos.y ), static_cast< float >( pos.z ) } );
+	mesh_node->SetRotation( { static_cast< float >( rot.x ), static_cast< float >( rot.y ), static_cast< float >( rot.z ) } );
+	mesh_node->SetScale( { static_cast< float >( scale[0] ), static_cast< float >( scale[1] ),static_cast< float >( scale[2] ) } );
+}
+
+auto getTransformFindAlgorithm( MFnTransform& transform)
+{
+	return [ &transform ]( std::pair<MObject, std::shared_ptr<wr::LightNode>> pair ) -> bool
+	{
+		MStatus status;
+		MFnLight fn_light( pair.first );
+		MFnDagNode dagnode = fn_light.parent( 0, &status );
+		MObject object = dagnode.object();
+		MFnTransform transform_rhs( dagnode.object(), &status );
+		
+		assert( status == MS::kSuccess );
+
+		if( transform.object() == transform_rhs.object() )
+		{
+			return true;
+		}
+		return false;
+	};
+}
+
+#pragma endregion
+
+#pragma region callbacks
+namespace wmr
+{
+	void AttributeLightTransformCallback( MNodeMessage::AttributeMessage msg, MPlug &plug, MPlug &other_plug, void *client_data )
+	{
+		// Check if attribute was set
+		if( !( msg & MNodeMessage::kAttributeSet ) )
+		{
+			return;
+		}
+
+		MStatus status = MS::kSuccess;
+		MFnTransform transform( plug.node(), &status );
+		if( status != MS::kSuccess )
+		{
+			MGlobal::displayInfo( status.errorString() );
+			return;
+		}
+		wmr::LightParser* light_parser = reinterpret_cast< wmr::LightParser* >( client_data );
+
+		// specialized find_if algorithm
+		auto it = std::find_if( light_parser->m_object_transform_vector.begin(), light_parser->m_object_transform_vector.end(), getTransformFindAlgorithm(transform) );
+		
+		MFnLight fn_light( it->first );
+		MFnDagNode dagnode = fn_light.parent( 0, &status );
+		MObject object = dagnode.object();
+		MFnTransform transform_rhs( dagnode.object(), &status );
+		if( transform_rhs.object() != transform.object() )
+		{
+			return; // find_if returns last element even if it is not a positive result
+		}
+
```

Comment:
Please have a look at your usage of unnecessary white spaces. It's just a bit inconsistent.